libsolidity/analysis/TypeChecker.cpp

```diff
@@ -355,7 +355,12 @@ void TypeChecker::checkFunctionOverride(FunctionDefinition const& function, Func
 		function.annotation().superFunction = &super;
 
 	if (function.visibility() != super.visibility())
+	{
+		// visibility is enforced to be external in interfaces, but a contract can override that with public
+		//if (super.isInInterface() && !function.isInInterface() && function.visibility() == FunctionDefinition::Visibility::Public)
```

Comment:
@chriseth I had this posted as a link on gitter and had a question: the above code would be the simplest way to do this, but it requires those helper and/or annotations in `FunctionDefinition`.

Is this a good way or do you have a better suggestion?

Reply:
You should be able to do something like `super.scope().contractKind() == Interface`

Reply:
Take a look at `bool TypeChecker::visit(FunctionDefinition const& _function)`. Perhaps we could add a single helper to the function that returns the ContractKind of its contract?

Reply:
In a helper? I think it also needs at least an assert that `super.scope()` is a `ContractDefinition`.

Reply:
Probably the most generic one is `function.inContractKind() == Interface` as that also replaces the `bool isLibraryFunction` in `visit(FunctionDefinition)`.

Reply:
an assert about `super.scope()` cannot hurt, yes.